### ISSUE TITLE
Move psi to a recursive array?

### ISSUE THREAD

**Issue by antoine-levitt** [JuliaMolSim/DFTK.jl#481]
Did we discuss this already?
https://github.com/SciML/RecursiveArrayTools.jl looks like a good way to store psi.

**Comment by mfherbst** [comment]
I don't think we did in this context and I agree, it nicely abstracts a few of the common patterns we face. Sounds like a good candidate to try.
